addons/base_report_engine_paper_muncher/engine/paper_muncher.py

```diff
@@ -0,0 +1,84 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import subprocess
+try:
+    import magic
+except ImportError:
+    magic = None
+
+from .http import HttpResponse, HttpRequest
+from .loader import OdooLoader
+from .utils import binary
+
+from odoo import _
+
+
+class PaperMuncher:
+
+    __slots__ = ('loader',)
+
+    loader: OdooLoader
+    binary: str = binary
+
+    def __init__(self, env, document, doc_name):
+        self.loader = OdooLoader(env, document, doc_name)
+
+    def _run(self, mode, *args: list[str]) -> tuple[bytes, str, Exception]:
```

Comment:
Just raise an exception when you have errors and avoid returning tuples here.

```suggestion
    def _run(self, mode: str, *args: str) -> bytes:
```